quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/scheduler/java/QuarkusSchedulerASTVisitor.java

```diff
@@ -0,0 +1,154 @@
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.Annotation;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.NormalAnnotation;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.java.validators.JavaASTValidator;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+import com.redhat.microprofile.jdt.internal.quarkus.QuarkusConstants;
+import com.redhat.microprofile.jdt.internal.quarkus.scheduler.SchedulerErrorCodes;
+import com.redhat.microprofile.jdt.internal.quarkus.scheduler.SchedulerUtils;
+
+public class QuarkusSchedulerASTVisitor extends JavaASTValidator {
+
+	private static Logger LOGGER = Logger.getLogger(QuarkusSchedulerASTVisitor.class.getName());
+
+	public QuarkusSchedulerASTVisitor() {
+		super();
+	}
+
+	@Override
+	public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		IJavaProject javaProject = context.getJavaProject();
+		return JDTTypeUtils.findType(javaProject, QuarkusConstants.SCHEDULED_ANNOTATION) != null;
+	}
+
+	@Override
+	public boolean visit(MethodDeclaration node) {
+		try {
+			List modifiers = node.modifiers();
+			for (Object modifier : modifiers) {
+				if (modifier instanceof Annotation) {
+					Annotation annotation = (Annotation) modifier;
+					if (isMatchAnnotation(annotation, QuarkusConstants.SCHEDULED_ANNOTATION)) {
+						try {
+							validateScheduledAnnotation(node, (NormalAnnotation) annotation);
+						} catch (ClassCastException e) {
+							;
+						}
+					}
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.WARNING, "An exception occurred when attempting to validate the annotation marked method");
+		}
+		super.visit(node);
+		return true;
+	}
+
+	/**
+	 * Checks if the given method declaration has a @Scheduled annotation, and if
+	 * so, provides diagnostics it's member(s)
+	 *
+	 * @param node       The method declaration to validate
+	 * @param annotation The @Scheduled annotation
+	 * @throws JavaModelException
+	 */
+	private void validateScheduledAnnotation(MethodDeclaration node, NormalAnnotation annotation)
+			throws JavaModelException {
+		Expression cronExpr = getAnnotationMemberValueExpression(annotation,
+				QuarkusConstants.SCHEDULED_ANNOTATION_CRON);
+		if (cronExpr != null) {
+			try {
+				String cronValue = cronExpr.resolveConstantExpressionValue().toString();
+				if (!malformedEnvDiagnostic(cronExpr, cronValue)) {
+					SchedulerErrorCodes cronPartFault = SchedulerUtils.validateCronPattern(cronValue);
+					if (cronPartFault != null) {
+						super.addDiagnostic(cronPartFault.getErrorMessage(), QuarkusConstants.QUARKUS_PREFIX, cronExpr,
+								cronPartFault, DiagnosticSeverity.Warning);
+					}
+				}
+			} catch (NullPointerException e) {
+				;
```

Comment:
You'll have to just check `cronExpr.resolveConstantExpressionValue()` is not null prior to using it. Also there is a better method to use than `toString()`. Have a look at the return value of this method call and see if you can find a better alternative.